src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx

```diff
@@ -379,9 +399,168 @@ export default class Oncoprinter extends React.Component<
         return WindowStore.size.width - 75;
     }
 
+    @action.bound
+    public handleSelectedClinicalTrackColorChange(
+        value: string,
+        color: RGBAColor | undefined
+    ) {
+        if (this.selectedClinicalTrack) {
+            this.props.store.setUserSelectedClinicalTrackColor(
+                this.selectedClinicalTrack.label,
+                value,
+                color
+            );
+        }
+    }
+
+    @observable trackKeySelectedForEdit: string | null = null;
+
+    @action.bound
+    setTrackKeySelectedForEdit(key: string | null) {
+        this.trackKeySelectedForEdit = key;
+    }
+
+    @computed get selectedClinicalTrack() {
+        return _.find(
+            this.props.store.clinicalTracks,
+            t => t.key === this.trackKeySelectedForEdit
+        );
+    }
+
+    @computed get defaultClinicalAttributeColoringForStringDatatype() {
+        if (this.selectedClinicalTrack) {
+            return _.mapValues(
+                getDefaultClinicalAttributeColoringForStringDatatype(
+                    this.selectedClinicalTrack.data
+                ),
+                hexToRGBA
+            );
+        }
+        return _.mapValues(RESERVED_CLINICAL_VALUE_COLORS, hexToRGBA);
+    }
+
+    @autobind
+    private getDefaultSelectedClinicalTrackColor(value: string) {
```

Comment:
maybe "getDefaultSelectedClinicalTrackColorForValue" just so it's clear that it's dependent on the value.  Also, maybe change name of value param to attributeValue.